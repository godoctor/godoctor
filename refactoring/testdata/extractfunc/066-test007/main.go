//<<<<<extract,8,2,17,2,newFunction,fail
package main

import "fmt"

func main() {
	x := 5
	for i := 0; i < 3; i++ {
		fn := func(key int) {
			fmt.Println("x is", x, "key is ", key)
			x = x + 20
			fmt.Println("the new value of x is ", x)
		}
		fn(333)
		x++
		fn(555)
	}
}
