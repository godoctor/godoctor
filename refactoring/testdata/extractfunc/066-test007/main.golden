//<<<<<extract,8,2,17,2,newFunction,fail
package main

import "fmt"

func main() {
	x := 5
	for i := 0; i < 3; i++ {
		fn := func() {
			fmt.Println("x is", x)
			x = x + 20
			fmt.Println("the new value of x is ", x)
		}
		fn()
		x++
		fn()
	}
}
